pytrip/vdx.py

```diff
@@ -288,10 +277,48 @@ def read_vdx(self, path):
                             i = v.read_vdx_old(content, i)
                 else:
                     i = v.read_vdx(content, i)
+                v.number = len(self.vois)
                 self.add_voi(v)
                 header_full = True
             i += 1
 
+        ctx_cube = getattr(self, 'cube', {})
+        dicom_data = getattr(ctx_cube, 'dicom_data', {})
+        data_datasets = getattr(dicom_data, 'data_datasets', {})
+        struct_data_dataset = data_datasets.get(AccompanyingDicomData.DataType.Struct, {})
+        ct_data_dataset = data_datasets.get(AccompanyingDicomData.DataType.CT, {})
+
+        # update voi number based on DICOM data
+        structure_set_roi_sequence = struct_data_dataset.get(Tag('StructureSetROISequence'), [])
+        dicom_voi_names = set([item.ROIName for item in structure_set_roi_sequence])
+        if structure_set_roi_sequence and (structure_set_roi_sequence.VM == self.number_of_vois()):
+            if dicom_voi_names == set(self.voi_names()):
+                for voi in self.vois:
+                    matched_item = [item for item in structure_set_roi_sequence if item.ROIName == voi.name]
```

Comment:
are you sure there can be only one? and if so, is there a way to do `find_first` method, to end loop when element is found?